winlogbeat/beater/eventlogger.go

```diff
@@ -133,14 +134,21 @@ func (e *eventLogger) run(
 runLoop:
 	for stop := false; !stop; {
 		err = api.Open(state)
-		if eventlog.IsRecoverable(err) {
-			e.log.Warnw("Open() encountered recoverable error. Trying again...", "error", err)
+
+		switch {
+		case eventlog.IsRecoverable(err):
+			e.log.Warnw("Open() encountered recoverable error. Trying again...", "error", err, "channel", api.Channel())
+			time.Sleep(time.Second * 5)
+			continue
+		case !api.IsFile() && eventlog.IsChannelNotFound(err):
+			e.log.Warnw("Open() encountered channel not found error. Trying again...", "error", err, "channel", api.Channel())
 			time.Sleep(time.Second * 5)
 			continue
```

Comment:
When running winlogbeat as a service under windows, the `continue` causes the service to never stop if requested to if a channel is not found. Is this expected behavior?